### PR TITLE
MULE-13003: Review mutability of Authentication object

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/security/Authentication.java [hunk 1]
```diff
@@ -10,24 +10,16 @@
...
 
   /**
-   * @param authenticated {@code true} if {@code this} {@link Authentication} has been authenticated
-   */
-  void setAuthenticated(boolean authenticated);
-
-  /**
-   * @return {@code true} if {@code this} {@link Authentication} has been authenticated
-   */
-  boolean isAuthenticated();
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/security/Authentication.java]: Why was this necessary?

Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/security/Authentication.java]: don't know, it was only used in assetions in tests, no usage in productive code.
